### ISSUE TITLE
Web: TextInput moves text cursor to end of input field after each keystroke

### ISSUE THREAD

**Issue by zypus** [JetBrains/compose-jb#1169]
TextInput currently only works correctly if the text cursor is at the end of the field. When the test cursor is instead somewhere inside of the text or at the start, the cursor is placed at the end of the input field on any text input.

Reproducible with the following minimal code:

```kotlin
fun main() {
    renderComposable(rootElementId = "root") {
        var text by remember { mutableStateOf("") }

        TextInput(text) {
            onInput {
                text = it.value
            }
        }
    }
}
```

**Comment by OliverO2** [comment]
The cause seems to be that Compose backfeeds the HTML input element's value on each input. Although the element's value does not change at the time, just setting it seems to make browsers consider the interaction finished and forget their selection state. 

As a workaround, you could
1. use replace `TextInput() { ... }` by its uncontrolled counterpart and
2. use a lambda to update its value only if it has changed:

```kotlin
Input(type = InputType.Text) {
    prop(
        { element: HTMLInputElement, value: String -> if (element.value != value) element.value = value },
        text
    )
    onInput { text = it.value }
}
```

**Comment by eymar** [comment]
It was fixed in 1.0.0-beta4

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-5269">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
